### ISSUE TITLE
Recurse subdirs?

### ISSUE THREAD

**Issue by midenok** [pauldreik/rdfind#37]
How to hardlink duplicates on whole filesystem?

**Comment by pauldreik** [comment]
Does using  / as path solve your problem?

**Comment by midenok** [comment]
I didn't try yet. But from help it is not clear whether it does recurse or not.

**Comment by midenok** [comment]
And how fast is the algorithm in that case? Whether it does checksum first and then sorts them?

**Comment by pauldreik** [comment]
> I didn't try yet. But from help it is not clear whether it does recurse or not.

You are right, the man page actually does not say that it does a recursive search! I will hijack this bug for this purpose.

> And how fast is the algorithm in that case? Whether it does checksum first and then sorts them?

This is described in https://github.com/pauldreik/rdfind#algorithm
The actual speed depends on how fast your filesystem is and how many duplicates you have. If all your files have distinct sizes, scanning is almost as fast as running ```find /```. If you have a lot of duplicate sizes, it has to read at least the first part of each file, which is a lot slower. And if you have lots of equally sized files which are or seem like duplicates, it will have to checksum the contents, which is as fast as ```sha1sum```.

**Comment by pauldreik** [comment]
added to the man page on the devel branch.
